pkg/registry/apps/rest/storage_apps.go

```diff
@@ -40,6 +41,15 @@ func (p RESTStorageProvider) NewRESTStorage(apiResourceConfigSource serverstorag
 	// If you add a version here, be sure to add an entry in `k8s.io/kubernetes/cmd/kube-apiserver/app/aggregator.go with specific priorities.
 	// TODO refactor the plumbing to provide the information in the APIGroupInfo
 
+	if extensionsGroupVersion := (schema.GroupVersion{Group: "extensions", Version: "v1beta1"}); legacyscheme.Registry.IsEnabledVersion(extensionsGroupVersion) {
+		if apiGroupInfo.SubresourceGroupVersionKind == nil {
+			apiGroupInfo.SubresourceGroupVersionKind = map[string]schema.GroupVersionKind{}
+		}
+		apiGroupInfo.SubresourceGroupVersionKind["deployments/scale"] = extensionsGroupVersion.WithKind("Scale")
+		apiGroupInfo.SubresourceGroupVersionKind["statefulsets/scale"] = extensionsGroupVersion.WithKind("Scale")
+		apiGroupInfo.SubresourceGroupVersionKind["replicasets/scale"] = extensionsGroupVersion.WithKind("Scale")
```

Comment:
This won't work, for a few reasons:

1. this forces `Scale` to be uniform in all versions of the `apps` API group. apps/v1beta1 and apps/v1beta2 accidentally shipped their own versions of Scale. We're trying to fix that in apps/v1, but that requires per-version control over subresource versioning. It's likely we should remove `SubresourceGroupVersionKind` and start using the `GroupVersionKindProvider` interface that lets each storage indicate the external version of the subresource.

2. we shipped releases with the scale subresources in `apps/v1beta1` and `apps/v1beta2` returning and requiring `apps/v1beta1` and `apps/v1beta2` versions of Scale. If you submit an `extensions/v1beta1` Scale object to a 1.8 apps/v1beta2 scale subresource, you get:

```json
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "the API version in the data (extensions/v1beta1) does not match the expected API version (apps/v1beta2)",
  "reason": "BadRequest",
  "code": 400
}
```

We cannot break wire compatibility with shipped releases. We'll need to figure out how to keep apps/v1beta1 and apps/v1beta2 behavior as-is, and fix it up for apps/v1